lib/graphql/batch/execution_strategy.rb

```diff
@@ -1,57 +1,57 @@
 module GraphQL::Batch
   class ExecutionStrategy < GraphQL::Query::SerialExecution
-    attr_reader :batched_queries
-
-    def initialize
-      @batched_queries = Hash.new{ |hash, key| hash[key] = [] }
-    end
-
     class OperationResolution < GraphQL::Query::SerialExecution::OperationResolution
       def result
-        result = super
-        until execution_strategy.batched_queries.empty?
-          queries = execution_strategy.batched_queries.shift.last
-          queries.first.class.execute(queries)
-        end
-        result
+        GraphQL::Batch::Promise.resolve(super).sync
+      ensure
+        GraphQL::Batch::Executor.current.clear
       end
     end
 
     class SelectionResolution < GraphQL::Query::SerialExecution::SelectionResolution
       def result
-        result_hash = super
-        result_hash.each do |key, value|
-          if value.is_a?(FieldResolution)
-            value.result_hash = result_hash
+        as_promise_unless_resolved(super)
+      end
+
+      private
+
+      def as_promise_unless_resolved(result)
+        all_promises = []
+        each_promise(result) do |obj, key, promise|
```

Comment:
Oh, it can't happen this is in `SelectionResolution` not `FieldResolution`, my mistake